### ISSUE TITLE
Address channel priority in installation instructions

### ISSUE THREAD

**Issue by maresb** [pymc-devs/pymc#6169]
The impetus comes from [this discourse thread](https://discourse.pymc.io/t/failure-to-link-with-mkl-rt-under-windows/10502).

I have the impression that the current leading cause of installation problems is mixed channels, in particular between `conda-forge` and `defaults`.

I personally tend to not have issues since in my `~/.condarc` I have

```yaml
channel_priority: strict
channels:
  - conda-forge
  - nodefaults
```

so to me the questions are 

* what sort of distribution of configurations do we see from people who end up with installation problems?
* how should we best recommend to resolve them?

The CF docs [recommend](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) to run

```bash
conda config --add channels conda-forge
conda config --set channel_priority strict
```

which adds `conda-forge` as the highest priority channel, and enables the modern `strict` channel priority setting.

A possible alternative is to add the `--strict-channel-priority` flag to any `conda`/`mamba` `create`/`install` commands.

I wonder if we should add a "Troubleshooting" section where we can perhaps elaborate a bit more about channels, as well as suggesting `mamba` for speeding up installation.

**Comment by aseyboldt** [comment]
I've also seen a couple of issues that were the result of people mixing conda-forge and defaults.
In a way this is more a problem of the conda ecosystem, not specifically pymc.
One problem with the "just use conda-forge" solution is that this applies to *all* environments in that conda setup, so it might change the behavior of totally unrelated projects. But there probably should be some mention of this in the docs :-)

**Comment by maresb** [comment]
I'd really like to understand which circumstances (namely conda distributions and possibly operating systems) lead to our standard installation command producing a mixed environment.

**Comment by aseyboldt** [comment]
It might just be that many users do `conda install -c conda-forge pymc` in a conda install with the default channel. I think that leaves you in a mixed environment.

**Comment by OriolAbril** [comment]
Closed by #6448, feel free to update the box to the commands straight away though as I commented on the PR.
